### PR TITLE
Add a dev release track

### LINKED ISSUE #58: [ADAP-749] [ADAP-718] Create a nightly/dev release track
To make it easier to test changes or get the absolutely latest changes from dbt-core or an adapter we want to have a nightly release that people can install from. 

This would have the tag of `nightly-release` and would _not_ be semantically versioned. Each night we will demote the current nightly release to `prior-nightly-release` and then create a new release tagged `nightly-release`. This means a consumer can simply install from the same release tag every night.

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #58 

Creates a new workflow to overwrite a dev bundle denoted as `0.0.1` every weekday. This bundle will pull from the main branch of each dbt-* package.

### CODE REVIEW

File: .github/workflows/release_dev_bundle.yml
Comment (emmyoop) [inline .github/workflows/release_dev_bundle.yml:1]: Why can we not use existing workflows and just send in the 0.0.0 version or preferably a "dev" flag of some sort?

File: release_creation/main.py [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
     create_new_draft_release_for_version,
     get_latest_bundle_release,
-    add_assets_to_release,
+    add_assets_to_release, create_dev_release,
 )
 from release_creation.bundle.create import generate_bundle
 from release_creation.release_logger import get_logger
 
 BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+DEV_VERSION = "0.0.0.0dev"
```
Comment (emmyoop) [inline release_creation/main.py:15]: why this pattern?

File: release_creation/main.py [hunk 2]
```diff
@@ -38,24 +40,37 @@ def main():
     args = parser.parse_args()
     version = args.input_version
     operation = args.operation
-    latest_version, is_draft, latest_release = get_latest_bundle_release(version)
-    logger.info(f"Retrieved latest version: {latest_version} "
-                f"and latest release: {latest_release.tag_name if latest_release else None}")
+    if version.startswith("0.0"):
+        latest_version = Version(DEV_VERSION)
+        is_draft = False
+        latest_release = None
```
Comment (emmyoop) [inline release_creation/main.py:46]: Won't this fail when there are multiple dev versions with the same tag?  Or will the create action just overwrite the previous release?  I would expect an error.

**Review by emmyoop — CHANGES_REQUESTED** [review]
This would benefit from README changes.

File: .github/workflows/release_dev_bundle.yml
Comment (colin-rogers-dbt) [reply .github/workflows/release_dev_bundle.yml:1]: Should have addressed this in the pr summary. My thinking was that the standard workflow added a couple things we don't need with a dev release that we can skip to save time: mac builds and draft release. Reconsidering both as I think running the standard process each night could be valuable in-and-of-itself (i.e. an early warning if something is broken with GHA) and once we add testing we can get some day-to-day performance metrics.

Comment (colin-rogers-dbt) [reply .github/workflows/release_dev_bundle.yml:1]: tl;dr you're right, the cons outweigh the pros on a separate workflow here

File: release_creation/main.py (hunk 2 above)
Comment (colin-rogers-dbt) [reply release_creation/main.py:46]: the create action overwrites the previous release

File: .github/workflows/release_dev_bundle.yml
Comment (emmyoop) [reply .github/workflows/release_dev_bundle.yml:1]: with the added benefit of not updating things in 2 places when updates are inevitably needed.

File: release_creation/main.py (hunk 1 above)
Comment (colin-rogers-dbt) [reply release_creation/main.py]: fixed!

File: .github/scripts/supported_python_versions.sh
Comment (emmyoop) [inline .github/scripts/supported_python_versions.sh:1]: I like pull out the dev versions specifically.  However the < 4 and >=4 python versions are now the same.  Are you leaving it in as a placeholder for when 3.11/spark is fixed?

File: .github/workflows/publish_draft_bundle.yml [hunk 3]
```diff
@@ -78,9 +78,10 @@ jobs:
 
       - name: "Set Python Versions"
         id: build-list
-        run: ./.github/scripts/supported_python_versions.sh ${{ steps.semver.outputs.minor }}
-
-      - name: Print Support of Python Versions Other Than 3.8
+        run: ./.github/scripts/supported_python_versions.sh \
+          ${{ steps.semver.outputs.major}} \
+          ${{ steps.semver.outputs.minor }}
+      - name: Print Supported Python Versions
```
Comment (emmyoop) [inline .github/workflows/publish_draft_bundle.yml:84]:
```suggestion
          ${{ steps.semver.outputs.minor }}

      - name: Print Supported Python Versions
```

File: release_creation/main.py [hunk 4]
```diff
@@ -38,33 +37,45 @@ def main():
     args = parser.parse_args()
     version = args.input_version
     operation = args.operation
-    latest_version, is_draft, latest_release = get_latest_bundle_release(version)
-    logger.info(f"Retrieved latest version: {latest_version} "
-                f"and latest release: {latest_release.tag_name if latest_release else None}")
+    if version.startswith("0.0"):
+        latest_version = Version.coerce(DEV_VERSION)
+        is_draft = True
```
Comment (emmyoop) [inline release_creation/main.py:42]: Does this need to be set?  In `create_dev_bundle` you have draft hardcoded to false.

**Review by emmyoop — APPROVED** [review]
Nothing blocking - just a few questions.

File: release_creation/main.py (hunk 4 above)
Comment (colin-rogers-dbt) [reply release_creation/main.py]: This should actually be hardcoded to false as we do a check later for draft is true

**Pull request merged**